MSBuildTasks/FilterTestingConfigurations.cs

```diff
@@ -0,0 +1,108 @@
+// the Apache License, Version 2.0 (the "License"); you may not use this 
+// file except in compliance with the License.  You may obtain a copy of the 
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the 
+// License for the specific language governing permissions and limitations
+// under the License.
+// 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Policy;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Remotion.BuildTools.MSBuildTasks
+{
+  public class FilterTestingConfigurations : Task
+  {
+    private readonly ITaskLogger _logger;
+    [Required] public ITaskItem[] Input { get; set; }
+    [Required] public ITaskItem[] ValidPlatforms { get; set; }
+    [Required] public ITaskItem[] ValidDatabaseSystems { get; set; }
+    [Required] public ITaskItem[] ValidBrowsers { get; set; }
+
+    [Output] public ITaskItem[] Output { get; set; }
+
+    public FilterTestingConfigurations ()
+    {
+      _logger = new TaskLogger (Log);
+    }
+
+    public FilterTestingConfigurations (ITaskLogger logger)
+    {
+      _logger = logger;
+    }
+
+    public override bool Execute ()
+    {
+      var validInputs = Input
+          .Where (HasValidPlatform)
+          .Where (HasValidDatabaseSystem)
+          .Where (HasValidBrowser);
+      Output = validInputs.ToArray();
+
+      _logger.LogMessage (
+          @"The following test configurations were ignored:
+{0}",
+          string.Join (Environment.NewLine, Input.Except (Output).Select (GetConfigurationString)));
+
+      var uniqueInputAssemblyNames = GetUniqueAssemblyNames (Input);
+      var uniqueOutputAssemblyNames = GetUniqueAssemblyNames (Output);
+      var untestedAssemblyNames = uniqueInputAssemblyNames.Where (assemblyName => !uniqueOutputAssemblyNames.Contains (assemblyName));
+      foreach (var assemblyName in untestedAssemblyNames)
+      {
+        _logger.LogWarning ($"All testing configurations in {assemblyName} were ignored.");
+      }
+
+      return true;
+    }
+
+    private string GetConfigurationString (ITaskItem taskItem)
+    {
+      var browser = taskItem.GetMetadata (TestingConfigurationMetadata.Browser);
+      var platform = taskItem.GetMetadata (TestingConfigurationMetadata.Platform);
+      var databaseSystem = taskItem.GetMetadata (TestingConfigurationMetadata.DatabaseSystem);
+      var executionRuntime = taskItem.GetMetadata (TestingConfigurationMetadata.ExecutionRuntime);
+      var configurationID = taskItem.GetMetadata (TestingConfigurationMetadata.ConfigurationID);
+      return $"{taskItem.ItemSpec}: {browser}, {databaseSystem}, {platform}, {executionRuntime}, {configurationID}";
```

Comment:
I thought the IDs serve the purpose to have an internal unique identifier on these configurations and externally, we would only expose the corresponding assembly. When I think about it though, it doesn't seem to matter that much anyway.

Reply:
Yeah, as the Configuration is pasted besides it, you are probably right. For a Log which could be used for Debugging, a unique Id seems better. For just reading and understanding, the itemspec seems better. Eh, lets see what MK thinks

Reply:
Spoke to MK, our ID **should** be the ItemSpec after all, and instead of an index it should be the project file name + the entire testing configuration string. FullPath and FileName will then be replaced with Metadata we define ourselves after all.

Reply:
ok that was a risk we took 😬 Sorry that you had to change it two times

Reply:
![image](https://user-images.githubusercontent.com/24362621/65498488-4424ef80-debc-11e9-9695-6ad109db604e.png)